### PR TITLE
(SERVER-2550) Miscellaneous fixes

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

File: src/clj/puppetlabs/puppetserver/certificate_authority.clj [hunk 1]
```diff
@@ -1463,17 +1463,24 @@
         cert-chain (utils/pem->certs cert-chain-path)
         ca-cert-key (utils/get-extension-value (first cert-chain)
                                                utils/subject-key-identifier-oid)
+        external-crl-chain (remove #(= ca-cert-key
+                                       (:key-identifier (utils/get-extension-value % utils/authority-key-identifier-oid)))
+                                   current-crls)
+        ca-crl (first (filter
+                       #(= ca-cert-key
+                           (:key-identifier (utils/get-extension-value % utils/authority-key-identifier-oid)))
+                       current-crls))
```
Comment (Magisus) [inline src/clj/puppetlabs/puppetserver/certificate_authority.clj:1472]: I'm still not sure this mess is better than doing the check in the loop. The key to this commit is dropping the logging.

Comment (Magisus) [reply src/clj/puppetlabs/puppetserver/certificate_authority.clj:1472]: We could cheat by just dropping the first CRL (which would make this cleaner), but I feel like we're trying not to perpetuate relying on the ordering? If that's true, should I find the cert by searching for the one with our "ca-name", or...?

**Comment by Magisus** [comment]
Doing manual validation with this.

File: src/clj/puppetlabs/puppetserver/certificate_authority.clj (hunk 1 above)
Comment (justinstoller) [reply src/clj/puppetlabs/puppetserver/certificate_authority.clj:1466]: fwiw, extracting the anonymous function might make it more readable?

**Comment by Magisus** [comment]
Travis is being slow and lame.

**Review by tvpartytonight — APPROVED** [review]

**Pull request merged**